Clarify DOTNET_ROOT documentation (#47059)
dotnet/docs: docs/core/tools/dotnet-environment-variables.md

- GitHub issue [dotnet/core#7699](https://github.com/dotnet/core/issues/7699)
- GitHub issue [dotnet/runtime#79237](https://github.com/dotnet/runtime/issues/79237)

> This environment variable is used only when running apps via generated executables (apphosts).

Author: Reworded. It might be better now?

- GitHub issue [dotnet/core#7699](https://github.com/dotnet/core/issues/7699)
- GitHub issue [dotnet/runtime#79237](https://github.com/dotnet/runtime/issues/79237)

These environment variables are used only when running apps via generated executables (apphosts). The order in which the environment variables are considered is:

1. `DOTNET_ROOT_<ARCH>`, where `<ARCH>` is the architecture of the running executable (apphost).

>     - For example, if running 64-bit executable on ARM64 OS or 64-bit OS, `DOTNET_ROOT_X64` is used. 

Reviewer: ```suggestion
    -  `DOTNET_ROOT_ARM64` is used for 64-bit process running on arm64 architecture.
    -  `DOTNET_ROOT_X64`  is used for 64-bit process running on x64 architecture.
    - `DOTNET_ROOT_X86`  is used for 32-bit process x86 architecture.
```

> 2. If running 32-bit executable on 64-bit OS, `DOTNET_ROOT(x86)` is considered.

Reviewer: ```suggestion
2. `DOTNET_ROOT(x86)` is used when running 32-bit process on 64-bit Windows. In other cases this environment variable is ignored.
```

> 3. `DOTNET_ROOT` is used as a fallback.

Reviewer 2: Not having the 'is used' seems to flow better with the lead-in of 'The order in which the environment variables are considered is:'
```suggestion
3. `DOTNET_ROOT`
```

- GitHub issue [dotnet/core#7699](https://github.com/dotnet/core/issues/7699)
- GitHub issue [dotnet/runtime#79237](https://github.com/dotnet/runtime/issues/79237)

These environment variables are used only when running apps via generated executables (apphosts). The order in which the environment variables are considered is:

1. `DOTNET_ROOT_<ARCH>`, where `<ARCH>` is the architecture of the running executable (apphost).

>     - For example: 

Author: ```suggestion
    - For example:
```

>         - `DOTNET_ROOT_ARM64` is used for an arm64 process.
>         - `DOTNET_ROOT_X64` is used for an x64 process. This may be running on x64 or arm64 architecture.
>         - `DOTNET_ROOT_X86` is used for an x86 process. This may be running on x86 or x64 architecture.

Reviewer 3: ```suggestion
     - `DOTNET_ROOT_ARM64` is used for an Arm64 process.
     - `DOTNET_ROOT_X64` is used for an x64 process. This process might be running on x64 or Arm64 architecture.
     - `DOTNET_ROOT_X86` is used for an x86 process. This process might be running on x86 or x64 architecture.
```

> 2. `DOTNET_ROOT(x86)` is used when running 32-bit process on 64-bit Windows. In other cases this environment variable is ignored.

Reviewer 3: ```suggestion
2. `DOTNET_ROOT(x86)` is used when a 32-bit process is running on 64-bit Windows. In other cases, this environment variable is ignored.
```

> 3. `DOTNET_ROOT`

Reviewer 3: ```suggestion
3. `DOTNET_ROOT`.
```
